mod_ood_proxy/lib/ood/nginx_stage.lua

```diff
@@ -74,12 +81,59 @@ end
 --[[
   capture2e
 
-  Give a string for stdin, get a string for merged stdout and stderr
+  Give a string for a command, get a string for merged stdout and stderr
 --]]
 function capture2e(cmd)
   return capture2(cmd .. " 2>&1")
 end
 
+--[[
+  capture2e_with_stdin
+
+  Give a command and some standard in, get a string for merged stdout and stderr
+--]]
+function capture2e_with_stdin(cmd, stdin)
+  local output_file = os.tmpname()
+  local redir_cmd = cmd .. " > " .. output_file .. " 2>&1"
+
+  local stdin_handle = io.popen(redir_cmd, "w")
+  stdin_handle:write(stdin)
+  output = stdin_handle:close()
+
+  local output_handle = io.open(output_file, "r")
+  output = output_handle:read("*all")
+  output_handle:close()
+
+  os.remove(output_file)
+
+  return output
```

Comment:
we will need to document that the pre hook when executed by Apache should not output any data to stdout or stderr since we report both as an error.